tests_integration/__tests__/doc-trim.js

```diff
@@ -0,0 +1,94 @@
+"use strict";
+
+const prettier = require("prettier/local");
+const docPrinter = prettier.doc.printer;
+const docBuilders = prettier.doc.builders;
+
+const printDocToString = docPrinter.printDocToString;
+const concat = docBuilders.concat;
+const line = docBuilders.line;
+const trim = docBuilders.trim;
+const group = docBuilders.group;
+const indent = docBuilders.indent;
+const markAsRoot = docBuilders.markAsRoot;
+
+// These tests don't use `runPrettier` because `trim` is not used by any
+// bundled parser (only third-party plugins).
+
+describe("trim", () => {
+  test.each([
+    [
+      // trims the current line
+      group(concat(["hello    ", trim])),
+      "hello"
+    ],
+    [
+      // trims existing indentation
+      group(
+        concat([
+          "function()",
+          line,
+          "{",
+          indent(
+            concat([
+              line,
+              group(concat([trim, "#if DEBUG"])),
+              line,
+              "alert(42);",
+              line,
+              group(concat([trim, "#endif"]))
+            ])
+          ),
+          line,
+          "}"
+        ])
+      ),
+      `function()
+{
+#if DEBUG
+  alert(42);
+#endif
+}`
+    ],
+    [
+      // trims up to the the first column, even if there's indented root.
+      group(
+        concat([
+          "{",
+          indent(
+            concat([
+              markAsRoot(line),
+              "foo:",
+              line,
+              "{",
+              indent(
+                concat([line, group(concat([trim, "// test"])), line, "42"])
+              ),
+              line,
+              "}"
+            ])
+          ),
+          line,
+          "}"
+        ])
+      ),
+      `{
+  foo:
+  {
+// test
+    42
+  }
+}`
+    ],
+    [
+      // ignores trimmed characters when fitting the line
+      group(concat(["hello  ", "  ", trim, line, "world!"])),
+      "hello world!"
+    ]
+  ])("trims correctly", (doc, expected) => {

```

Comment:
We could use [the `printf`-like parameters provided by jest](https://jestjs.io/docs/en/api#1-testeachtable-name-fn-timeout) here to improve our test names:

```js
test.each([
  [
    "test name", // change those comments into test names
    doc,
    expected,
  ]
])("%s", (_, doc, expected) => {
//  ^^    ^
})
```

```
 PASS  tests_integration/__tests__/doc-trim.js
    ✓ trims the current line
    ✓ trims existing indentation
```

instead of 

```
 PASS  tests_integration/__tests__/doc-trim.js
    ✓ trims correctly
    ✓ trims correctly
```